implementation/leprechaun/src/shared/api/http-exceptions.ts

```diff
@@ -0,0 +1,62 @@
+import { HttpException, HttpStatus } from '@nestjs/common';
+import { ApiProperty } from '@nestjs/swagger';
+
+export const ErrorKind: Record<string, [string, number]> = {
+  SchemaMismatchException: ['schema-mismatch-exception', HttpStatus.UNPROCESSABLE_ENTITY],
+  NotFoundException: ['resource-not-found-exception', HttpStatus.NOT_FOUND],
+  InternalServerError: ['internal-server-error', HttpStatus.INTERNAL_SERVER_ERROR],
+} as const;
+
+type Values<Object> = Object[keyof Object];
+type Keys<Object> = keyof Object;
+
+interface AppError {
+  kind: string;
+  code: number;
+  messageEn: string;
+  messagePl?: string;
+}
+
+export class HttpExceptionDto implements AppError {
+  @ApiProperty({ description: 'Error kind', enum: () => Object.values(ErrorKind), isArray: false })
```

Comment:
**suggestion:** `enum` can take an array as value, is there any reason to do
```
() => Object.values(ErrorKind)
```
instead of
```
Object.values(ErrorKind)
```
?

(applies to multiple places, but I'll only leave a comment here)